app/models/rook.rb

```diff
@@ -1,5 +1,21 @@
 class Rook < Piece
 
+  def valid_move?(new_x, new_y)
+    position_exist?(new_x, new_y) && passes_rook_rules?(new_x, new_y)
+  end
+
+  def position_exist?(x, y)
+    x >= 0 && x <= 7 && y >= 0 && y <= 7
+  end
+
```

Comment:
That's very true, thanks for pointing this out. I'll try switching `position_exist?` to the the Piece model